src/Features/CSharp/Portable/NavigationBar/CSharpNavigationBarItemService.cs

```diff
@@ -73,18 +73,27 @@ public CSharpNavigationBarItemService()
                         continue;
                     }
 
-                    var method = member as IMethodSymbol;
-                    if (method != null && method.PartialImplementationPart != null)
+                    if (member is IMethodSymbol { PartialImplementationPart: { } } methodSymbol)
```

Comment:
Without this change, navigation bar still works.
Before:
![image](https://github.com/dotnet/roslyn/assets/24360909/b500655a-9e55-43cc-bab1-03f9d2402770)

After:
![image](https://github.com/dotnet/roslyn/assets/24360909/e7e1b838-1495-46a6-8e03-fc26efe73219)

After behavior matches the existing behavior of partial method.